application/event_processor/dos.py

```diff
@@ -39,86 +43,127 @@ class DoSService:
     ]
 
     def __init__(self, db_cursor_row):
-
-        # Sets the attributes of this object to those found in the db row
+        """Sets the attributes of this object to those found in the db row
+        Args:
+            db_cursor_row (dict): Change Request changes
+        """
         for i, attribute_name in enumerate(self.db_columns):
-            attrbute_value = db_cursor_row[i]
-            setattr(self, attribute_name, attrbute_value)
+            attribute_value = db_cursor_row[i]
+            setattr(self, attribute_name, attribute_value)
 
-    def __repr__(self):
+    def __repr__(self) -> str:
+        """Returns a string representation of this object"""
         if self.publicname is not None:
             name = self.publicname
         elif self.name is not None:
             name = self.name
         else:
             name = "NO-VALID-NAME"
 
-        return f"<uid={self.uid} ods={self.odscode} type={self.typeid} " f"status={self.statusid} name='{name}'>"
+        return f"<uid={self.uid} ods={self.odscode} type={self.typeid} status={self.statusid} name='{name}'>"
 
-    def ods5(self):
-        """First 5 digits of odscode"""
-        return self.odscode[0:5]
-
-    def get_changes(self, nhs_entity) -> dict:
+    def get_changes(self, nhs_entity: NHSEntity) -> dict:
         """Returns a dict of the changes that are required to get
         the service inline with the given nhs_entity
         """
-
         changes = {}
+        add_field_to_change_request_if_not_equal(changes, "website", self.web, nhs_entity.Website)
+        add_field_to_change_request_if_not_equal(changes, "postcode", self.postcode, nhs_entity.Postcode)
+        add_field_to_change_request_if_not_equal(changes, "phone", self.publicphone, nhs_entity.Phone)
+        add_field_to_change_request_if_not_equal(changes, "publicname", self.publicname, nhs_entity.OrganisationName)
+        add_address_to_change_request_if_not_equal(changes, "address", self.address, nhs_entity)
+        return changes
 
-        # WEBSITE
-        if self.web != nhs_entity.Website:
-            changes["Website"] = nhs_entity.Website
 
-        # TODO in future tickets: Add in checks for the rest of the
-        # possible changes
+def add_field_to_change_request_if_not_equal(changes: dict, change_key: str, dos_value: str, nhs_uk_value: str) -> dict:
+    """Adds field to the change request if the field is not equal
+    Args:
+        changes (dict): Change Request changes
+        change_key (str): Key to add to the change request
+        dos_value (str): Field from the DoS database for comparision
+        nhs_uk_value (str): NHS UK Entity value for comparision
 
-        return changes
+    Returns:
+        dict: Change Request changes
+    """
+    logger.debug(f"Comparing {change_key} values")
+    if compare_values(str(dos_value), str(nhs_uk_value)):
```

Comment:
`str(dos_value) != str(nhs_uk_value)` is easier to read, is there a reason for renaming the this equality check? Is it just the logging, would consider moving the logging into the consumer

Reply:
I can this log line but there is an equivalent log comment in line 135 of this file